include/bringauto/fleet_protocol/http_client/FleetApiClient.hpp

```diff
@@ -0,0 +1,103 @@
+#pragma once
+
+#include "CppRestOpenAPIClient/api/CarApi.h"
+#include "CppRestOpenAPIClient/api/DeviceApi.h"
+#include "CppRestOpenAPIClient/api/ModuleApi.h"
+#include "RequestFrequencyGuard.hpp"
+
+#include <optional>
+
+
+namespace bringauto::fleet_protocol::http_client {
+
+
+class FleetApiClient
+{
+public:
+    /**
+     * @brief Constructs the fleet API client
+     * @param apiUrl URL of Fleet v2 HTTP API
+     * @param apiKey api key used for authorization
+     * @param companyName company identifier in the API
+     * @param carName car identifier in the API
+     * @param maxRequestsThresholdCount max amount of allowed requests in the specified time period (only for getCommands/getStatuses)
+     * @param maxRequestsThresholdPeriodMs time period in which the amount of requests is checked (only for getCommands/getStatuses)
+     * @param delayAfterThresholdReachedMs time to sleep for in ms when threshold is initially reached (only for getCommands/getStatuses)
+     * @param retryRequestsDelayMs delay in ms between requests until request rate is no longer over the threshold (only for getCommands/getStatuses)
+     */
+    FleetApiClient(std::string apiUrl, std::string apiKey, std::string companyName, std::string carName,
+                   int32_t maxRequestsThresholdCount = 10, int32_t maxRequestsThresholdPeriodMs = 5000,
+                   int32_t delayAfterThresholdReachedMs = 5000, int32_t retryRequestsDelayMs = 200);
+
+    ~FleetApiClient();
+
+    /**
+     * @brief Sets the DeviceId parameters, required for sendCommand and sendStatuse functions
+     * @param moduleId moduleId of device
+     * @param deviceType type of device
+     * @param deviceRole role of device
+     * @param deviceName name of device
+     */
+    void setDeviceIdentification(int32_t moduleId, int32_t deviceType, std::string deviceRole, std::string deviceName);
```

Comment:
use const std::string & to avoid copy

Reply:
all function string params are now const &